### ISSUE TITLE
Racy test

### ISSUE THREAD

**Issue by foolswood** [concert/clapi#12]
```
protocolServer graceful interrupt: [Failed]
ERROR: Prelude.undefined
CallStack (from HasCallStack):
  error, called at libraries/base/GHC/Err.hs:79:14 in base:GHC.Err
  undefined, called at test/TestServer.hs:188:34 in main:TestServer
```

**Comment by ch3pjw** [comment]
`while stack test --ta --select-tests=gracefu; do echo yay; done`

**Comment by foolswood** [comment]
Looks like you found it.

**Comment by ch3pjw** [comment]
i.e. I think this was fixed by #14
